src/components/navigation/header/menu/menu.tsx

```diff
@@ -0,0 +1,24 @@
+import * as NavigationMenu from '@radix-ui/react-navigation-menu'
+import { useEffect, useState } from 'react'
+
+import { AuthItems } from './auth-items'
+import { Item } from './item'
+import { useInstanceData } from '@/contexts/instance-context'
+
+export function Menu() {
+  const { headerData } = useInstanceData()
+  const [mounted, setMounted] = useState(false)
+
+  useEffect(() => setMounted(true), [])
```

Comment:
Why do you have this construction?

Reply:
We render in two passes.
First there is SSR that assumes no authed user.
We hide the "user" part of the menu until we run this component on the client and have the user data available.